source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -244,6 +272,14 @@ JsonTranscoderConfig::methodToRequestInfo(const Protobuf::MethodDescriptor* meth
   return ProtobufUtil::Status();
 }
 
+ProtobufUtil::Status
+JsonTranscoderConfig::translateProtoMessageToJson(const Protobuf::Message& message,
```

Comment:
Use Protobuf::util::MessageToJsonString so you don't have to serialize to string once, and respect the `Protobuf::util::JsonPrintOptions print_options_` for that.

Reply:
We should also respect the user-provided protobuf descriptor, as `google.rpc.Status` has this `repeated Any details` field, which may contain arbitrary user types, e.g. those standard details types from [google/rpc/error_details.proto](https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto).

Thus `MessageToJsonString` would call the same `BinaryToJsonString` with global (built-in) TypeResolver, and fail to convert the message.

The `GrpcJsonTranscoderFilterConvertGrpcStatusTest.TranscodingStatusFromBinaryHeaderWithDetails` can reproduce this problem as it decodes a Status which contains the helloworld.HelloReply type in `details`, and the transcoder's test binary doesn't have helloworld_proto_cc in dependencies.

Reply:
Speaking of those types in error_details.proto,
in the JsonTranscoderConfig ctor we can add types to a user-provided protobuf descriptor, as it's done with the google.rpc.Status type, and actually it adds whole proto files (Protobuf::FileDescriptorProto structures). 
So we could do the same with those somewhat "standard" details types, build them into envoy binary and add them to user's descriptor. This would help users to avoid writing messages like this:
```
message Dummy {
  google.rpc.RequestInfo request_info = 1;
}
```

Reply:
But it already uses `print_options_`.

Adding error_details.proto would be great, but I think it would better be done in a separate PR.